app/src/src_nonfreenet/org/breezyweather/sources/meteoam/MeteoAmResultConverter.kt

```diff
@@ -0,0 +1,213 @@
+/**
+ * This file is part of Breezy Weather.
+ *
+ * Breezy Weather is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, version 3 of the License.
+ *
+ * Breezy Weather is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.breezyweather.sources.meteoam
+
+import breezyweather.domain.location.model.Location
+import breezyweather.domain.weather.model.Current
+import breezyweather.domain.weather.model.Daily
+import breezyweather.domain.weather.model.HalfDay
+import breezyweather.domain.weather.model.PrecipitationProbability
+import breezyweather.domain.weather.model.Temperature
+import breezyweather.domain.weather.model.WeatherCode
+import breezyweather.domain.weather.model.Wind
+import breezyweather.domain.weather.wrappers.HourlyWrapper
+import breezyweather.domain.weather.wrappers.WeatherWrapper
+import org.breezyweather.common.exceptions.InvalidOrIncompleteDataException
+import org.breezyweather.sources.meteoam.json.MeteoAmForecastData
+import org.breezyweather.sources.meteoam.json.MeteoAmForecastResult
+import org.breezyweather.sources.meteoam.json.MeteoAmForecastStats
+import org.breezyweather.sources.meteoam.json.MeteoAmObservationDatasets
+import org.breezyweather.sources.meteoam.json.MeteoAmObservationResult
+import org.breezyweather.sources.meteoam.json.MeteoAmReverseLocation
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+fun convert(
+    location: Location,
+    reverseLocation: MeteoAmReverseLocation,
+    timezone: String
+): Location {
+    return location.copy(
+        timeZone = timezone,
+        country = reverseLocation.country!!,
+        countryCode = reverseLocation.country_code,
+        province = reverseLocation.county,
+        city = reverseLocation.city!!
+    )
+}
+
+fun convert(
+    forecastResult: MeteoAmForecastResult,
+    observationResult: MeteoAmObservationResult
+): WeatherWrapper {
+    val timeseries = forecastResult.timeseries
+    val params = forecastResult.paramlist
+    val stats = forecastResult.extrainfo?.stats
+    val data = forecastResult.datasets?.data
+    val oParams = observationResult.paramlist
+    val observation = observationResult.datasets?.get("0")
+
+    if (timeseries == null || params == null || stats == null || data == null || oParams == null) {
+        throw InvalidOrIncompleteDataException()
+    }
+
+    if (params.size < 10 || params[0] != "2t" || params[1] != "r" || params[2] != "pmsl" || params[3] != "tpp"
```

Comment:
Is it really that predictable? Can't we do something more "generic" that could take any parameter in any order?

Reply:
Reworked the code so that `getCurrent` and `getHourlyForecast` can take parameters in arbitrary order rather than fixed order.